src/Features/Core/Portable/Workspace/MiscellaneousFileUtilities.cs

```diff
@@ -46,6 +46,12 @@ internal static ProjectInfo CreateMiscellaneousProjectInfoForDocument(
             compilationOptions = GetCompilationOptionsWithScriptReferenceResolvers(services, compilationOptions, filePath);
         }
 
+        if (parseOptions != null && fileExtension != languageInformation.ScriptExtension)
+        {
+            // Any non-script misc file should not complain about usage of '#:' ignored directives.
+            parseOptions = parseOptions.WithFeatures([.. parseOptions.Features, new("FileBasedProgram", "true")]);
```

Comment:
This should probably move into the CSharpSyntaxTreeFactoryService's GetDefaultParseOptionsWithLatestLanguageVersion(). That's also used by metadata as source, which if that also gets used for Source Link, we'd want this since we don't want errors if a source linked file had this too.

(As it stands this would also be setting this for VB, which may not be ideal.)

Reply:
I think setting it for VB doesn't affect anything. It is just not recognized/used by the VB compiler.

It also seems like no diagnostics are expected for MetadataAsSource or SourceLink, since diagnostics are not actionable in those contexts.

I am happy to make the suggested change, though.